src/main/scala/tech/cryptonomic/conseil/tezos/ApiOperations.scala

```diff
@@ -167,151 +267,52 @@ object ApiOperations {
+    (action: Query[Tables.OperationGroups, Tables.OperationGroupsRow, Seq]) extends Action
 
-  private def getFilterLimit(filter: Filter): Int = if (filter.limit.isDefined) filter.limit.get else 10
-
-  // End helper functions for constructing Slick queries
 
+  case class AccountsAction(action: Query[Tables.Accounts, Tables.AccountsRow, Seq]) extends Action
 
   /**
     * Filters Accounts, Operation Groups, Operations, and Blocks tables based on users input to the filter.
     * @param filter Filter parameters.
     * @return
     */
   private def getFilteredTables(filter: Filter): Try[FilteredTables] = {
-    fetchLatestBlock().flatMap { latestBlock =>
-      fetchMaxBlockLevelForAccounts().flatMap { maxLevelForAccounts =>
-        Try {
-          val filteredAccounts = Tables.Accounts.filter(account =>
-            filterAccountIDs(filter, account) &&
-              filterAccountDelegates(filter, account) &&
-              filterAccountManagers(filter, account) &&
-              account.blockLevel === maxLevelForAccounts)
-
-          val filteredOpGroups = Tables.OperationGroups.filter({ opGroup =>
-            filterOperationIDs(filter, opGroup)
-          })
-
-          val filteredOps = Tables.Operations.filter({ op =>
-            filterOperationKinds(filter, op) &&
-              filterOperationDestinations(filter, op) &&
-              filterOperationSources(filter, op)
-          })
-
-          val filteredBlocks = Tables.Blocks.filter({ block =>
-            filterBlockIDs(filter, block) &&
-              filterBlockLevels(filter, block) &&
-              filterChainIDs(filter, block) &&
-              filterProtocols(filter, block)
-          })
-
-          FilteredTables(filteredAccounts, filteredBlocks, filteredOpGroups, filteredOps)
-        }
-      }
+
+    def filterOn(maxLevel: BigDecimal): FilteredTables = {
+      val filteredAccounts = Tables.Accounts.filter(account =>
+        filterAccountIDs(filter, account) &&
+          filterAccountDelegates(filter, account) &&
+          filterAccountManagers(filter, account) &&
+          account.blockLevel === maxLevel)
+
+      val filteredOpGroups = Tables.OperationGroups.filter({ opGroup =>
+        filterOperationIDs(filter, opGroup)
+      })
+
+      val filteredOps = Tables.Operations.filter({ op =>
+        filterOperationKinds(filter, op) &&
+          filterOperationDestinations(filter, op) &&
+          filterOperationSources(filter, op)
+      })
+
+      val filteredBlocks = Tables.Blocks.filter({ block =>
+        filterBlockIDs(filter, block) &&
+          filterBlockLevels(filter, block) &&
+          filterChainIDs(filter, block) &&
+          filterProtocols(filter, block)
+      })
+
+      FilteredTables(filteredAccounts, filteredBlocks, filteredOpGroups, filteredOps)
     }
+
+    for {
+      _ <- fetchLatestBlock() // is this needed?
+      maxLevelForAccounts <- fetchMaxBlockLevelForAccounts()
```

Comment:
I'll have to double check the codebase, but if fetchMaxBlockLevelForAccounts still works without the latest block in scope from fetchLatestBlock, then we should be fine

Reply:
Technically they can't influence each other, and if they do I would take it as a design shortcoming and try to change that to make it explicit.

The fetch is done on db for both operations, so I won't expect any side-effect from the first call to influence the second one.

The only semantic reason might be that we want to be sure that a _latest block exists on the DB_... but even if it doesn't, the max level for accounts would simply be the default, and I can't see anything wrong with that.